File: xy10674/backend/src/models/OperationLog.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const OperationLog = sequelize.define('OperationLog', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  user_id: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  operation: {
    type: DataTypes.STRING(50),
    allowNull: false
  },
  module: {
    type: DataTypes.STRING(50),
    allowNull: false
  },
  record_id: {
    type: DataTypes.INTEGER
  },
  detail: {
    type: DataTypes.TEXT
  },
  ip_address: {
    type: DataTypes.STRING(50)
  },
  user_agent: {
    type: DataTypes.STRING(255)
  }
}, {
  tableName: 'operation_logs',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: false
});

module.exports = OperationLog;

File: zy71975/backend/app/models/operation_log.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, Boolean, ForeignKey
from sqlalchemy.orm import relationship

from app.core.database import Base


class OperationLog(Base):
    __tablename__ = "operation_logs"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    meeting_id = Column(Integer, ForeignKey("meetings.id"), comment="会议ID")
    operation_type = Column(String(50), nullable=False, comment="操作类型: upload, compare, correct, export, delete")
    operation_detail = Column(Text, comment="操作详情")
    operator = Column(String(100), comment="操作人")
    ip_address = Column(String(50), comment="IP地址")
    status = Column(String(50), default="success", comment="状态: success, failed")
    error_message = Column(Text, comment="错误信息")

    is_deleted = Column(Boolean, default=False, comment="是否删除")
    created_at = Column(DateTime, default=datetime.now, comment="创建时间")

    meeting = relationship("Meeting", back_populates="operation_logs")
